fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AlterTableClauseAnalyzer.java

```diff
@@ -1119,21 +1129,13 @@ private void analyzeAddPartition(OlapTable olapTable, List<PartitionDesc> partit
                 RangePartitionInfo rangePartitionInfo = (RangePartitionInfo) partitionInfo;
                 SingleRangePartitionDesc singleRangePartitionDesc = ((SingleRangePartitionDesc) partitionDesc);
                 singleRangePartitionDesc.analyze(rangePartitionInfo.getPartitionColumnsSize(), cloneProperties);
-                if (!existPartitionNameSet.contains(singleRangePartitionDesc.getPartitionName())) {
-                    rangePartitionInfo.checkAndCreateRange(table.getIdToColumn(), singleRangePartitionDesc,
```

Comment:
Does this function contain modification logic? I didn't find it. It seems that there is a sorting problem, so adding a temp clone object can solve it. The verification logic should be left in the analyze stage, which helps us keep clear logic.